packages/cactus-plugin-ledger-connector-corda/src/main-server/kotlin/gen/kotlin-spring/src/main/kotlin/org/hyperledger/cactus/plugin/ledger/connector/corda/server/impl/ApiPluginLedgerConnectorCordaServiceImpl.kt

```diff
@@ -346,4 +352,108 @@ class ApiPluginLedgerConnectorCordaServiceImpl(
+            }
+        }
+
+        logger.info("Monitoring for state '{}' started.", stateName)
+        return StartMonitorV1Response(true)
+    }
+
+    /**
+     * Read all transactions that were not read by any consumer yet.
+     * Must be called after startMonitorV1 and before stopMonitorV1.
+     * Transactions buffer must be explicitly cleared with clearMonitorTransactionsV1
+     */
+    override fun getMonitorTransactionsV1(req: GetMonitorTransactionsV1Request?): GetMonitorTransactionsV1Response {
+        val stateName = req?.stateFullClassName
+
+        if (stateName.isNullOrEmpty()) {
+            // Missing input class name
+            logger.info("Request rejected because missing state class name")
+            return GetMonitorTransactionsV1Response("", listOf())
+        }
+
+        val transactions = this.watchedTransactionsBuffer.getOrElse(stateName) { listOf<GetMonitorTransactionsV1ResponseTx>() }
+        logger.debug("Returning {} transaction to the caller", transactions.size)
+        return GetMonitorTransactionsV1Response(stateName, transactions)
+    }
+
+    /**
+     * Clear monitored transactions based on index from internal buffer.
+     * Any future call to getMonitorTransactionsV1 will not return transactions removed by this call.
+     */
+    override fun clearMonitorTransactionsV1(req: ClearMonitorTransactionsV1Request?): ClearMonitorTransactionsV1Response {
+        val stateName = req?.stateFullClassName
+        if (stateName.isNullOrEmpty()) {
+            // Missing input class name
+            logger.info("Request rejected because missing state class name")
+            return ClearMonitorTransactionsV1Response(false)
+        }
+
+        val indexesToRemove = req?.txIndexes
+        if (indexesToRemove.isNullOrEmpty()) {
+            logger.info("No indexes to remove")
+            return ClearMonitorTransactionsV1Response(true)
+        }
+
+        val transactions = this.watchedTransactionsBuffer.getOrElse(stateName) { mutableListOf<GetMonitorTransactionsV1ResponseTx>() }
+        logger.debug("Transactions before remove", transactions.size)
+        transactions.removeAll { it.index in indexesToRemove }
+        logger.debug("Transactions after remove", transactions.size)
+        return ClearMonitorTransactionsV1Response(true)
+    }
+
+    /**
+     * Stop monitoring state changes for stateClass specified in request body.
+     * Removes all transactions that were not read yet, unsubscribes from the monitor.
+     */
+    override fun stopMonitorV1(req: StopMonitorV1Request?): StopMonitorV1Response {
+        val stateName = req?.stateFullClassName
+
+        if (stateName.isNullOrEmpty()) {
+            // Missing input class name
+            logger.info("Request rejected because missing state class name")
+            return StopMonitorV1Response(false)
+        }
+
+        this.watchedTransactionsSubs[stateName]?.unsubscribe()
```

Comment:
@outSH Is there some logic in here that gracefully frees up resources if the client didn't do so itself? 

Example: 
1. Browser client starts monitoring
2. Server creates a subscription here 
3. Browser client disappears (phone battery dies, laptop falls into a river, etc.)
4. Server keeps on chugging along with the subscription or does it automatically cleans up at some point?

Reply:
@petermetz And what if BLP looses a connection for few minutes due to network error and we trash the events he was expecting to receive to proceed with it's logic? It could be devastating from business point of view. To have this at least once delivery behavior we need some sort of queue on the connector. We can clear it from time to time to prevent memory draining, put the queue in redis or regular DB, but the more improvements in this area we do, the more we're reinventing the wheel of a message broker. In current implementation I've assumed long-lived clients only (BLP) that would sooner or later reconnect, and would like to fetch any state changes that occurred during it's blackout.

This whole problem comes down to a question - **what are expected message delivery guarantees** for block, or any other monitored async data?  We're right now clarifying exact requirements for message delivery guarantees we need internally, **we would like to know your point of view on this as well**.
- Current socketio monitoring (for instance, in besu) is not reliable at all, since blocks can be easily missed, which might be OK in some cases, but in others (like finance area where corda majors) is definitely not.
- The more guarantees we need, the more backend implementation and communication complexity we need.
- At some point it will be better to just use an actual message broker like kafka or AMQP based solution.
- Another thing is, that actual enterprise-grade message broker or anything near it will not work well with your idea ApiClients in browser, mobile phones, toaster ovens talking to the connector directly :)
- So, please share the requirements and possibly scenarios you must support, and then we can try to come to some conclusion. I will implement the required changes in this PR then.

Reply:
@outSH Delivery guarantees are a different rabbit hole. I just want to know if in its current state it's leaking memory or not. :-)

For the delivery guarantees: I largely agree. My point of view is that we can probably support different transports for the same endpoints, meaning that pushing the data down the initiating web socket would be one transport, appending to a queue could be another, AMQP, JMS or whatever we can actually support.

Reply:
@outSH 

> @petermetz Yeah, the question about delivery guarantees was not directly related to this review but to whole PR, I wanted to clarify expectation for the messaging implementation :)

Fair enough. We could open a dedicated issue in the tracker, link to this conversation for more context and then hash it out further if needed (not much to say form my side though, we appear to be on the same page)

> @petermetz I'd personally prefer to settle on single communication mechanism (at least for async), I'm afraid having multitude of pluggable technologies can turn into matinanace nightmare as soon as we have any production workloads using them. Nevertheless, our teams don't have any specific requirements at the moment, so I think we can go with free-for-all approach at the moment.

I also don't want to maintain a several different transports, but I do want to be able to say `"Yes"` to someone who comes and asks us: `Hey, is there a way to add support for a different transport without having to majorly refactor the entire codebase?` Then they would have to implement & maintain that for themselves on account of their own production workloads (them = the OSS community)
This is also not a change request just me dumping thoughts.

Reply:
@outSH 

> @petermetz As for memory leak, in current version each client that started a monitoring on the connector is checked periodically, and clients who has no activity for specified amount of time are removed and they resources are freed. Timeout is configurable.

Thank you very much, marking it as resolved!